src/ccalls.jl

```diff
@@ -227,7 +234,7 @@ end
 function _call_gap_func(func::GapObj)
     fptr = GET_FUNC_PTR(func, 0)
     ret = ccall(fptr, Ptr{Cvoid}, (Ptr{Cvoid},), pointer_from_objref(func))
-    return _GAP_TO_JULIA(ret)
+    _GAP_TO_JULIA(ret)
```

Comment:
Perhaps this is more Julia idomatic (though I still find it confusing); in any case, if we make this change, then I suggest we make it in _all_ `_call_gap_func` methods at once. And perhaps not in this PR? Unless it is relevant somehow for it?

Reply:
Okay, this is actually an accident. I had started wrapping the `_call_gap_func()` bodies in `@sync` until I realized that this was unnecessary, but apparently had forgotten to revert the change to the first of them. We can revert this one, too.